File: 17solve/17-154.pas
// Автор В.Н. Шубинкин
##
var a := ReadLines('17-1.txt').Select(t -> t.ToInteger).ToArray;
var count := 0;
var minDist := 10000;
var pos := -1;
for var i := 1 to a.High - 1 do
begin
  if (a[i] > a[i - 1]) and (a[i] > a[i + 1]) then
  begin
    count += 1;
    if pos <> -1 then minDist := min(minDist, i - pos);
    pos := i
  end;
end;
print(count, minDist)

File: 17solve/17-154.py
# Автор В.Н. Шубинкин

with open('17-1.txt') as f:
    a = [int(x) for x in f.readlines()]

count = 0
min_dist = 20000
pos = -1
for i in range(1, len(a) - 1):
    if a[i - 1] < a[i] > a[i + 1]:
        count += 1
        if pos != -1:
            min_dist = min(min_dist, i - pos)
        pos = i
print(count, min_dist)
